app/src/main/java/org/simlar/service/liblinphone/LinphoneHandler.java

```diff
@@ -377,4 +383,150 @@ private static void enableDebugMode(final boolean enabled)
 	{
 		LinphoneCoreFactory.instance().setDebugMode(enabled, "DEBUG");
 	}
+
+	public void requestVideoUpdate(final boolean enable)
+	{
+		Lg.i("requestVideoUpdate: enable=", enable);
+
+		final LinphoneCall currentCall = getCurrentCall();
+		if (currentCall == null) {
+			Lg.w("no current call to add video to");
+			return;
+		}
+
+		final LinphoneCallParams params = currentCall.getCurrentParams();
+		if (params.getVideoEnabled() == enable) {
+			Lg.i("requestVideoUpdate already: ", enable, " => aborting");
+			return;
+		}
+
+		params.setVideoEnabled(enable);
+		mLinphoneCore.updateCall(currentCall, params);
+	}
+
+	public void preventAutoAnswer()
+	{
+		Lg.i("preventAutoAnswer");
+
+		final LinphoneCall currentCall = getCurrentCall();
+		if (currentCall == null) {
+			Lg.w("no current call to prevent auto answer for");
+			return;
+		}
+
+		try {
+			mLinphoneCore.deferCallUpdate(currentCall);
+		} catch (final LinphoneCoreException e) {
+			Lg.ex(e, "LinphoneCoreException during deferCallUpdate");
+		}
+	}
+
+	public void acceptVideoUpdate(final boolean accept)
+	{
+		Lg.i("acceptVideoUpdate accept=", accept);
+
+		final LinphoneCall currentCall = getCurrentCall();
+		if (currentCall == null) {
+			Lg.w("no current call to accept video for");
+			return;
+		}
+
+		final LinphoneCallParams params = currentCall.getCurrentParams();
+		if (accept) {
+			params.setVideoEnabled(true);
+		}
+
+		try {
+			mLinphoneCore.acceptCallUpdate(currentCall, params);
+		} catch (final LinphoneCoreException e) {
+			Lg.ex(e, "LinphoneCoreException during acceptCallUpdate");
+		}
+	}
+
+	public void setVideoWindow(final Object videoWindow)
+	{
+		Lg.i("setVideoWindow");
+
+		if (mLinphoneCore == null) {
+			Lg.e("setVideoWindow: mLinphoneCore is null => aborting");
+			return;
+		}
+
+		mLinphoneCore.setVideoWindow(videoWindow);
+	}
+
+	public void setVideoPreviewWindow(final Object videoPreviewWindow)
+	{
+		Lg.i("setVideoPreviewWindow");
+
+		if (mLinphoneCore == null) {
+			Lg.e("setVideoPreviewWindow: mLinphoneCore is null => aborting");
+			return;
+		}
+
+		enableCamera(videoPreviewWindow != null);
+		mLinphoneCore.setPreviewWindow(videoPreviewWindow);
+	}
+
+	private void setFrontCameraAsDefault()
+	{
+		int cameraId = 0;
+		for (final AndroidCameraConfiguration.AndroidCamera androidCamera : AndroidCameraConfiguration.retrieveCameras()) {
+			if (androidCamera.frontFacing) {
+				cameraId = androidCamera.id;
+			}
+		}
+
+		mLinphoneCore.setVideoDevice(cameraId);
```

Comment:
If no camera was found this, I would expect an error.

Reply:
I think it is better to use another camera than. But warning makes sense for me:
edd0d8d1d5f84c438f82f3f293d494b16b80a261

And we should abort if there is no camera at all:
ed85f2719ff348c4676cc88c2a6f2f52dc6f8a89